src/common/dump.cpp

```diff
@@ -44,6 +44,44 @@ void dump_to_xflash(void) {
     w25x_fetch_error();
 }
 
+void dump_err_to_xflash(const char *error, const char *title) {
+    dumpmessage_t dump_message;
+    w25x_sector_erase(w25x_error_start_adress);
+    dump_message.flag = 0;
+    strlcpy(dump_message.msg_title, title, sizeof(dump_message.msg_title));
+    strlcpy(dump_message.msg_content, error, sizeof(dump_message.msg_content));
+    w25x_program(w25x_error_start_adress, (uint8_t *)(&dump_message), sizeof(dumpmessage_t));
+    w25x_fetch_error();
+}
+
+int dump_in_xflash_get_message(char *msg_dst, char *title_dst) {
+    dumpmessage_t dump_message;
+    w25x_rd_data(w25x_error_start_adress, (uint8_t *)(&dump_message), sizeof(dumpmessage_t));
+    if (w25x_fetch_error())
+        return 0;
+    strlcpy(title_dst, dump_message.msg_title, sizeof(dump_message.msg_title));
+    strlcpy(msg_dst, dump_message.msg_content, sizeof(dump_message.msg_content));
+    return 1;
+}
+
+void dump_in_xflash_clear_message_flag(void) {
+    uint8_t dumpmessage_flag;
+    w25x_rd_data(w25x_error_start_adress, &dumpmessage_flag, 1);
```

Comment:
This is suspitious. As dumpmessage_t.flag is defined as unsigned short earlier.

Reply:
Mistake, flag is now defined as uint8_t in `7c21391`